src/main/java/de/tudarmstadt/ukp/jwktl/db/berkeley/BerkeleyConfigurationModel.java

```diff
@@ -0,0 +1,250 @@
+	public BerkeleyConfigurationModel() throws Exception {
+		super();
+		classMap = new HashMap<String, ClassMetadata>();
+		entityMap = new HashMap<String, EntityInfo>();
+
+		addPersistentMetadata(Pronunciation.class.getName());
+		addPersistentMetadata(Quotation.class.getName());
+		addPersistentMetadata(WikiString.class.getName());
+		addPersistentMetadata(WiktionaryEntry.class.getName());
+		addPersistentMetadata(WiktionaryExample.class.getName());
+		addPersistentMetadata(WiktionaryRelation.class.getName());
+		addPersistentMetadata(WiktionarySense.class.getName());
+		addPersistentMetadata(WiktionaryTranslation.class.getName());
+		addPersistentMetadata(WiktionaryWordForm.class.getName());
+
+		addClassMetadata(WiktionaryPage.class.getName());
+	}
+
+	@Override
+	public synchronized ClassMetadata getClassMetadata(String className) {
+		ClassMetadata classMetadata = super.getClassMetadata(className);
+		if (classMetadata == null) {
+			classMetadata = classMap.get(className);
+		}
+		return classMetadata;
+	}
+
+	@Override
+	public synchronized EntityMetadata getEntityMetadata(String className) {
+		EntityMetadata entityMetadata = super.getEntityMetadata(className);
+		if (entityMetadata == null) {
+			/* Return the collected entity metadata. */
+			EntityInfo info = entityMap.get(className);
+			if (info != null)
+				entityMetadata = new EntityMetadata(className, info.priKey, Collections.unmodifiableMap(info.secKeys));
+
+		}
+		return entityMetadata;
+	}
+
+	private void addPersistentMetadata(String className) {
+		ClassMetadata metadata = new ClassMetadata(className, 0, null, false, null, null, null, null);
+		classMap.put(className, metadata);
+		/* Add any new information about entities. */
+		updateEntityInfo(metadata);
+	}
+
+	private ClassMetadata addClassMetadata(String className) throws Exception {
+		boolean isEntity = true;
+		int version = 0;
+		String proxiedClassName = null;
+
+		ClassMetadata metadata;
+		Class<?> type;
+		try {
+			type = resolveClass(className);
+		} catch (ClassNotFoundException e) {
+			return null;
+		}
+
+		/* Get instance fields. */
+		List<Field> fields = new ArrayList<Field>();
+		boolean nonDefaultRules = getInstanceFields(fields, type);
+		Collection<FieldMetadata> nonDefaultFields = null;
+		if (nonDefaultRules) {
+			nonDefaultFields = new ArrayList<FieldMetadata>(fields.size());
+			for (Field field : fields) {
+				nonDefaultFields.add(new FieldMetadata(field.getName(), field.getType().getName(), type.getName()));
+			}
+			nonDefaultFields = Collections.unmodifiableCollection(nonDefaultFields);
+		}
+
+		Field foundField = type.getDeclaredField("id");
+		PrimaryKeyMetadata pkMD = new PrimaryKeyMetadata(foundField.getName(), foundField.getType().getName(), type.getName(), null);
+
+		Map<String, SecondaryKeyMetadata> mapSK = new HashMap<String, SecondaryKeyMetadata>();
+
+		Field field = WiktionaryPage.class.getDeclaredField("title");
+		String elemClassName = null; // getElementClass(field);

```

Comment:
this method could be cleaned up a little. why does WiktionaryPage.class gets passed in a String, then transformed into a class, and in this later referred to again, hardcoded?